coordinator-tool/src/bin/election-coordinator.rs

```diff
@@ -1186,50 +1277,99 @@ async fn handle_initial_weights(
     let traverse_config = indexer::TraverseConfig::new_single(endpoint, first_block.block_height);
     let (sender, mut receiver) = tokio::sync::mpsc::channel(20);
     let cancel_handle = tokio::spawn(traverse_config.traverse(indexer::BlockEventsIndexer, sender));
-    while let Some((block, normal, specials)) = receiver.recv().await {
-        if block.block_slot_time > end {
+    let mut affected = BTreeSet::new();
+
+    'blocks: while let Some((block, normal, specials)) = receiver.recv().await {
+        if block.block_height > last_block.block_height {
             drop(receiver);
             eprintln!("Done indexing");
             break;
         }
         bar.set_message(block.block_slot_time.to_string());
         bar.inc(1);
-        let mut affected = BTreeSet::new();
+
+        // First collect the affected addresses within the payday
         for tx in normal {
             for addr in tx.affected_addresses() {
                 affected.insert(AccountAddressEq::from(addr));
             }
         }
-        for special in specials {
+        for special in &specials {
             for addr in special.affected_addresses() {
                 affected.insert(AccountAddressEq::from(addr));
             }
         }
-        let block_ident = BlockIdentifier::from(block.block_height);
-        for acc in affected {
-            let info = client
-                .get_account_info(&AccountAddress::from(acc).into(), block_ident)
-                .await?;
-            let index = info.response.account_index.index as usize;
+
+        // Check if the block is a payday block by going through the special transaction
+        // events. If not, then skip the block.
+        for special in &specials {
+            let has_payday_event = matches!(
+                special,
+                SpecialTransactionOutcome::PaydayPoolReward { .. }
+                    | SpecialTransactionOutcome::PaydayAccountReward { .. }
+                    | SpecialTransactionOutcome::PaydayFoundationReward { .. }
+            );
+
+            if !has_payday_event {
```

Comment:
I think you are skipping the block if just any of the special transaction outcomes is not a payday event.
Instead, I believe we should check whether none of the special transaction outcomes are a payday event.

Reply:
Fixed, good catch!